### ISSUE TITLE
Reading samples directly at runtime (port to STM32)

### ISSUE THREAD

**Issue by Tronix286** [schellingb/TinySoundFont#74]
Given: 
STM32 micro with ~384Kb-512Kb RAM;
SD card connected to STM32 with fast SDIO 6-wire interface running at 24MHz clock;
FatFS, so files at SD-card can operate fopen() fclose() fseek() etc...

In your opinion, is it possible to read SF2 samples not into the main memory, but directly in pieces of the required size from a file (stream) at the runtime of sound rendering? Or into a small buffer?
Is it difficult to implement?
Thanks in advance.

**Comment by yishengjiang99** [comment]
I don’t think you want to do any file io ops in real time.. see #4 in this article https://atastypixel.com/four-common-mistakes-in-audio-development/

**Comment by maks** [comment]
This is quite an old issue but @yishengjiang99 given the existence of at least 2 successful streaming-from-sdcard grooveboxes in the form of the Deluge and the M8 I think the correct answer is that it is possible given a sufficiently careful implementation.
